controllers/openshift_controller.go

```diff
@@ -1035,9 +942,75 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 		r.Log.Info("SCNodeRole is: " + machinePool)
 	}
 
-	wasMcJustCreated, err := r.createExtensionMc(machinePool)
-	if err != nil {
-		return ctrl.Result{Requeue: true}, nil
+	// Based on LayeredImageDeployment feature, we'll either create an ExtensionMachineConfig or
+	// just apply the image
+	wasMcJustCreated := false
+
+	// If featuregate LayeredImageDeployment is enabled then create the image based MC
+	// If it's disabled, create the extension based MC
+
+	if r.FeatureGatesStatus[featuregates.LayeredImageDeployment] {
+		r.Log.Info("PreviousMcName and PreviousMethod, ImageBasedDeployment=true",
+			"PreviousMcName", r.deploymentState.PreviousMcName, "PreviousMethod", r.deploymentState.PreviousMethod)
+		// Check if LayeredImageDeployment config is provided
+		// If not log an error and return
+		if !r.FeatureGates.IsFeatureConfigMapPresent(context.TODO(), featuregates.LayeredImageDeployment) {
+			r.Log.Info("LayeredImageDeployment feature is enabled but no configuration provided")
+			err := fmt.Errorf("LayeredImageDeployment feature is enabled but no configuration provided")
+			return ctrl.Result{Requeue: true}, err
+		}
+
+		// Check if the previous deployment method was different
+		// When we first time enter this condition the PreviousMethod will be empty
+		if r.deploymentState.PreviousMethod != "image" {
+			// Rollback the previous deployment
+			if r.deploymentState.PreviousMcName != "" {
+				// Delete the previous MachineConfig
```

Comment:
It seems that we should just check for the existence of the `extension_mc_name`  MC and delete it if it exists.